components/automate-ui/src/app/modules/infra-proxy/tree-table/component/tree-table.component.html

```diff
@@ -0,0 +1,25 @@
+<table mat-table [dataSource]="dataSource" [ngClass]="formatElevation()">
+  <div *ngFor="let column of displayedColumns; first as isFirst;">
+    <ng-container matColumnDef="{{column}}">
+      <th mat-header-cell *matHeaderCellDef [ngClass]="{'vertical-separator': options.verticalSeparator}">
+        {{options.capitalisedHeader ? (column | titlecase) : column}}
+      </th>
+      <td mat-cell *matCellDef="let element" [ngClass]="{'vertical-separator': options.verticalSeparator}">
+        <div *ngIf="isFirst">
+          <div class="value-cell">
+            <div [class]="'child-step-'+(element.depth)"></div>
```

Comment:
Even though it adds a little complexity to the markup, I like this solution a lot better so that we're not constantly firing that function.  Thank you! 🎉